book-cw/part-2-coding/regex.adoc

```diff
@@ -1,3 +1,348 @@
 = Coding in Regex
 
-Chapter content removed for re-review (in progress).
+The patterning syntax known as _regular expressions_ or _regex_ is as notorious for its alleged steep learning curve as for its confirmed usefulness.
+Nobody seriously questions the _efficacy_ of this set of scripting rules, but the efficiency of writing regular expressions is sometimes unclear.
+Knowing when they're essential takes some familiarity, so we'll consider a few use=case types.
+
+Regular expressions can be used used not only to scan for predefined patterns in arbitrary text; expressions can extract contextualized data from within those matched patterns, passing it to an application for further manipulation and storage.
+
+The concept of regular expressions is notoriously difficult to learn.
+I admit I didn't appreciate their potential for the first many years I begrudgingly used them to solve otherwise impossible data conversion and other IT tasks.
+Appreciating regex can make a world of difference to learning regex.
+
+If you have never or barely used regex, perhaps it is about to become your first coding language.
+If you've used it before but never taken the time to learn its full potential, this chapter should be helpful as well.
+
+Seasoned regex coders can keep me honest.
+This book is open source; help me bring the power of this intimidating but mercifully straightforward scripting syntax to struggling documentarians everywhere.
+
+.What-ex?
+[NOTE]
+I pronounce the shortened term _REJ-ex_, though I believe I've heard more developers use _REGG-ex_.
+Or maybe it just strikes me more starkly when I hear it pronounced differently than I say it.
+
+== What is Regex?
+
+A regular expression is a way of describing a pattern using common characters in order to search for the described pattern in another body of characters.
+The target can be pretty much any form of text, such as a conventional document or a flattened dataset.
+A regex parsing engine uses these expressions to evaluate target texts for matches.
+
+Put differently by the regex-focused site Rexegg, “A regex is a text string that describes a pattern that a regex engine uses in order to find text (or positions) in a body of text, typically for the purposes of validating, finding, replacing, or splitting.”
+
+At its simplest application, think of regex as a much smarter version of the `*` or `%` wildcard symbols you've probably used in other contexts.
+Where a conventional wildcard finds _any and all text_, regex can find shockingly detailed patterns without false positives.
+
+The string `codewriting.org` is matched by the following regex patterns:
```

Comment:
Recommend that you either summarize these three regexes or say that you'll explain them later, otherwise beginners aren't going to see why that gibberish of symbols is useful here. Regexes are really hard to read and parse if you didn't write them, and in this case you're writing for complete beginners. I'd run away screaming from that first one as a beginner.